### PR TITLE
test(e2e): Add `ssh localhost` test

### PR DESCRIPTION
**Pull request by moduli**
This PR adds another test case that starts a `boundary connect` followed by a `ssh localhost -p PORT` to connect to the target. This is another common way that users access targets.

### CODE REVIEW

File: testing/internal/e2e/tests/static/connect_localhost_test.go [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+	ctxCancel, cancel := context.WithCancel(context.Background())
+	port := "12345"
+	cmdChan := make(chan *e2e.CommandResult)
+	go func() {
+		t.Log("Starting session...")
+		cmdChan <- e2e.RunCommand(ctxCancel, "boundary",
+			e2e.WithArgs(
+				"connect",
+				"-target-id", newTargetId,
+				"-listen-port", port,
```
Comment (moduli) [inline testing/internal/e2e/tests/static/connect_localhost_test.go:45]: I couldn't figure out a way to access the output of this command without canceling it (in order to get the port that gets generated), so I had to specify a port here.

Comment (johanbrandhorst) [reply testing/internal/e2e/tests/static/connect_localhost_test.go:45]: If you wanted to, I think you could do this with a pipe, but this seems fine and much less complicated.

**Review by johanbrandhorst — APPROVED** [review]
